src/generator.cs

```diff
@@ -1785,7 +1785,7 @@ public TrampolineInfo MakeTrampoline (Type t)
 		}
 
 		var rt = mi.ReturnType;
-		var rts = IsNativeEnum (rt) ? "var" : rt.ToString ();
```

Comment:
Yes, without this fix you will get an issue with the trampolines that use generics because you get the class name like NSArra'1 rather than NSArray<String>

Reply:
Ideally this would be in a separate PR to make blaming it in the future more obvious.

Consider calling it out with explaination in the PR/commit description for possible future us.

Reply:
This can be a diff fix. The issue where is that the rt.ToString() ois doing the wrong thing and is generating a non-compilable name. I'll add this as a separate branch with a decent explanation.